include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -815,19 +817,46 @@ struct __simple_write_to_idx
 };
 
 template <typename _Predicate>
-struct __gen_count_pred
+struct __gen_mask
+{
+    template <typename _InRng>
+    bool
+    operator()(_InRng&& __in_rng, std::size_t __idx) const
+    {
+        return __pred(__in_rng[__idx]);
```

Comment:
Do we need to forward `__in_rng` here?

Reply:
After second thought, I don't think it would buy us anything to forward the range here just to call `operator[]` on it, so disregard this comment.

Reply:
Yeah, I questioned this myself earlier in the previous PR.  I think we *can* do it, but I don't think we have to and I don't think we get much from it.